Model/Cron/Customers.php

```diff
@@ -142,16 +149,40 @@ public function getSubscribersBatch($limit, $offset, $last_update)
      * Changes customer subscription status to unsubscribed in Magento database.
      *
      * @param array $unsubscribers_list List of unsubscribers emails from Smaily
+     * @param int|string ID of website
      * @return void
      */
-    public function removeUnsubscribers($unsubscribers_list)
+    public function removeUnsubscribers($unsubscribers_list, $websiteId)
     {
         $table = $this->connection->getTableName('newsletter_subscriber');
+        $storeIds = $this->getAllStoreIdsForWebsite($websiteId);
 
         foreach ($unsubscribers_list as $unsubscriber_email) {
-            $query = "UPDATE $table SET subscriber_status = '0' WHERE subscriber_email = :UNSUBSCRIBER_EMAIL";
-            $binds = ['UNSUBSCRIBER_EMAIL' => $unsubscriber_email];
+            $query = "UPDATE $table SET subscriber_status = '0' WHERE subscriber_email = :UNSUBSCRIBER_EMAIL
+                AND store_id IN (:STORE_IDS)";
+            $binds = ['UNSUBSCRIBER_EMAIL' => $unsubscriber_email, 'STORE_IDS' => implode(',', $storeIds)];
             $this->connection->query($query, $binds);
         }
     }
+
+    /**
+     * Get all Store IDs listed under website, return them in an array.
+     *
+     * @param string Website ID
+     * @return array[int] Store IDs in array
+     */
+    private function getAllStoreIdsForWebsite($websiteId)
+    {
+        $stringStoreIds = $this->storeWebsiteRelation->getStoreByWebsiteId($websiteId);
+
+        // Convert to int. Simpler for using in SQL binds.
+        $intStoreIds = array_map(
+            function ($value) {
+                return (int)$value;
+            },
+            $stringStoreIds
+        );
```

Comment:
You can also write this as:
```suggestion
        $intStoreIds = array_map('intval', $stringStoreIds);
```

Using anonymous functions in loops can be inefficient, because the function is always recreated. Using an existing function is better in some cases. In this case you can assume that number of stores per website is quite low and it really doesn't matter if you are using anonymous functions or not.

Reply:
Should I replace the anonymous function in `array_filter` too? (Data.php:406). Then `SUBSCRIBERS_BATCH_LIMIT` (which is 1000) plays a role

Reply:
It would be more efficient, but when looking at time saved on the whole synchronization process, it does not save a lot of time. It is up to you if you want to change it or not.

Reply:
I couldn't find good sources about how inefficient anonymous functions are in time. I would leave it be now